skale/contracts/schains.py

```diff
@@ -0,0 +1,91 @@
+#   -*- coding: utf-8 -*-
+#
+#   This file is part of SKALE.py
+#
+#   Copyright (C) 2019-Present SKALE Labs
+#
+#   SKALE.py is free software: you can redistribute it and/or modify
+#   it under the terms of the GNU Affero General Public License as published by
+#   the Free Software Foundation, either version 3 of the License, or
+#   (at your option) any later version.
+#
+#   SKALE.py is distributed in the hope that it will be useful,
+#   but WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#   GNU Affero General Public License for more details.
+#
+#   You should have received a copy of the GNU Affero General Public License
+#   along with SKALE.py.  If not, see <https://www.gnu.org/licenses/>.
+""" Schains.sol functions """
+
+from Crypto.Hash import keccak
+
+from skale.contracts import BaseContract
+from skale.utils.helper import format_fields
+
+
+FIELDS = [
+    'name', 'owner', 'indexInOwnerList', 'partOfNode', 'lifetime', 'startDate', 'startBlock',
+    'deposit', 'index', 'chainId'
+]
+
+
+class SChains(BaseContract):
+
+    def get_schains_internal(self):
```

Comment:
I think we should use cached property here

Reply:
done